### PR TITLE
VDS-904 Implement indexation of members User Groups (member tags)

### PR DESCRIPTION
**Pull request by trueboroda**
Also implement turning on logic by feature flag state.

### CODE REVIEW

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Module.cs [hunk 1]
```diff
@@ -181,14 +188,38 @@ public void PostInitialize(IApplicationBuilder appBuilder)
             var demoFeaturesSection = configuration.GetSection("DemoFeatures");
             featureStorage.AddHighPriorityFeatureDefinition(demoFeaturesSection);
 
-            featureStorage.TryAddFeature("ConfigurableProduct", true);
-            featureStorage.TryAddFeature("UserGroupsInheritance", "Developers");
+            featureStorage.TryAddFeature(ModuleConstants.Features.ConfigurableProduct, true);
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Module.cs]: Please specify exactly the module in which constants are used.

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Module.cs [hunk 2]
```diff
@@ -181,14 +188,38 @@ public void PostInitialize(IApplicationBuilder appBuilder)
...
             inProcessBus.RegisterHandler<DemoProductPartChangedEvent>(async (message, token) => await appBuilder.ApplicationServices.GetService<LogChangesProductPartsHandler>().Handle(message));
             inProcessBus.RegisterHandler<DemoTaggedMemberChangedEvent>(async (message, token) => await appBuilder.ApplicationServices.GetService<LogChangesTaggedMembersHandler>().Handle(message));
 
+            var featureManager = appBuilder.ApplicationServices.GetRequiredService<IFeatureManager>();
+
+            var userGroupsInheritanceFeatureIsEnabled = featureManager.IsEnabledAsync(ModuleConstants.Features.UserGroupsInheritance).GetAwaiter().GetResult();
+
+            #region Search
+
+            var documentIndexingConfigurations = appBuilder.ApplicationServices.GetRequiredService<IEnumerable<IndexDocumentConfiguration>>();
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Web/Module.cs:207]: Add empty line

File: src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Services/Customer/DemoMemberService.cs [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+                    member.Groups = tags.IsNullOrEmpty() ? member.Groups : tags;
+                }
+            }
+
+            return members;
+        }
+
+        public override async Task SaveChangesAsync(Member[] members)
+        {
+            var userGroupsInheritanceFeatureIsEnabled = await _featureManager.IsEnabledAsync(ModuleConstants.Features.UserGroupsInheritance);
```
Comment (pushnitsa) [inline src/VirtoCommerce.DemoSolutionFeaturesModule.Data/Services/Customer/DemoMemberService.cs]: Add empty line

**Review by pushnitsa — CHANGES_REQUESTED** [review]

**Review by pushnitsa — APPROVED** [review]

**Comment by trueboroda** [comment]
It will bring with this PR https://github.com/VirtoCommerce/vc-module-demo-features/pull/73

**Pull request closed without merging**